### ISSUE TITLE
Max-height:100% and overflow:hidden prevent scrolling in mobile Safari and Voiceover

### ISSUE THREAD

**Issue by marcysutton** [angular/material#2315]
Due to a [bug in mobile Safari/Webkit](https://bugs.webkit.org/show_bug.cgi?id=141893) and Voiceover, pages with `max-height: 100%; overflow: hidden;` on the body cannot be scrolled using 3-finger swipe (tab order swiping works ok). Content is always announced as "page 1 of 1", even though there is clearly more scrollable content. This most notably affects the sidenav in the docs app, which has a lot of overflow scrolling. 

Fixing this would be a pretty major style change, so we'll have to talk as a team about the best approach. I know we talked about removing these styles a while back, so this should be another reason to consider it. Related: #681
